src/types.ts

```diff
@@ -15,7 +15,6 @@ export const argoCDAppDetails = t.type({
     history: t.array(t.type({
       id: t.number,
       revision: t.string,
-      deployStartedAt: t.string,
       deployedAt: t.string,

```

Comment:
I'm wondering how this io-ts library will handle the case where deployedAt is undefined. Will it fail trying to decode?

Reply:
@iain-b Think you got a point there, it caused issues with deployStartedAt, might do the same with deployedAt. I am removing it as I think io-ts doesn't support optional props but for future reference we might reconsider using io-ts library

Reply:
You can do this https://github.com/gcanti/io-ts/blob/master/Decoder.md#the-intersect-combinator to have optional properties but I agree it's really ugly for what is quite a common use case. I agree that we might reconsider.

Reply:
I did follow that pattern but it still has weird behaviour when handling with undefined values. I did however found a way with using _union_. Maybe that would be good for now?